### ISSUE TITLE
Fix WWDRCA certificate.

### ISSUE THREAD

**Issue by Fengson** [travis-ci/travis-ci#7065]
I am having this issue:
Whenever I try to run Travis for iOS (Xcode 8.2) I get this error:
`Fix WWDRCA Certificate`

This makes my builds to fail, as I am using Fastlane's match to sign the app.

I've read this thread: https://github.com/travis-ci/travis-ci/issues/6935
but it didn't help me. Maybe this line:
`security set-key-partition-list -S apple-tool:,apple: -s -k keychainPass keychainName`
should be entered before every script? How can I make it so my Security keychain contains updated WWDRCA certificate?

I've tried deleting it and installing new one, as suggested here: 
https://github.com/fastlane-old/gym/issues/100#issuecomment-184335220

Is there a way Travis would be updated to use new WWDRCA certificate by default?

**Comment by BanzaiMan** [comment]
I do not think this is an error, per se, although it is quite confusing.

`Fix WWDRCA Certificate` message comes from our build script, and it does not indicate any error.

The message `Unable to delete certificate matching …` indicates that our attempt to remove the offending certificate failed (because it doesn't exist).

The last message `security: AppleWWDRCA.cer: already in /Library/Keychains/System.keychain` means that the attempt to add the correct certificate also failed (because it already exists).

In short, we should skip this step if we don't have to.

**Comment by BanzaiMan** [comment]
I don't think these error messages are related to the hang, by the way.

**Comment by Fengson** [comment]
The error message from math says: "Check in Keychain Access for an expired WWDR certificate: http://stackoverflow.com/a/35409835/774 has more info."

How can I make sure Travis uses updated WWDRCA certificate? Without it, build fails.

**Comment by BanzaiMan** [comment]
@Fengson Thanks for pointing that out. I overlooked it. We will look into it.

**Comment by cotsog** [comment]
@Fengson: can you send over your `Fastfile` and `./fastlane/travis.sh` script to support [at] travis-ci [dot] com? Please remove any confidential information these files may contain. Thanks!

**Comment by Fengson** [comment]
@cotsog Email has been sent. Thank you for your help.

**Comment by aterribili** [comment]
Any news?

**Comment by guoyingtao** [comment]
Any news?

**Comment by cotsog** [comment]
@aterribili, @guoyingtao: FYI, @Fengson's problem was the issue documented here: https://docs.travis-ci.com/user/common-build-problems/#Mac%3A-macOS-Sierra-(10.12)-Code-Signing-Errors.

Please reach out at support [at] travis-ci [dot] com if you have a different issue. Make sure to include your repository name and a link to a recent build showing the problem.

Thanks!

**Comment by guoyingtao** [comment]
@cotsog Thanks for your answer.
I am still a little confused about some contend of the document, like:

To fix this issue, you will need to add the following command after you have imported your certificate:
security set-key-partition-list -S apple-tool:,apple: -s -k keychainPass keychainName

My quesition is
1. Where should I add the "command"? In .travis.yml?
2. Where to set keychainPass and keychainName?

Thanks!

**Comment by cotsog** [comment]
@guoyingtao: Happy to answer your questions below

1) Often, customers building their iOS app on Travis CI will have a Bash script taking care of importing their certificate. Hence, you would put this command at the end of this script. You can see an example script [in this tutorial](https://www.objc.io/issues/6-build-tools/travis-ci/#add-scripts).

2) The password and name of your keychain can be set using the `security create-keychain` command e.g.

```
security create-keychain -p travis ios-build.keychain
```

Hope this helps!

**Comment by guoyingtao** [comment]
@cotsog If I only use iphonesimulator for my test, do I still need Bash script to import my certificate.

**Comment by guoyingtao** [comment]
@costog I mean If I only want to test on simulator, I don't think I need to sign my app. But I still meet the fix WWDRCA problem. Does it mean if I choose ios8.1 or above, signing app os forced no matter I use simulator or not? Thanks a lot!

**Comment by cotsog** [comment]
@guoyingtao: I believe you don't need to import certificates if you just want to run your tests on the simulator.

To have a better understanding of your situation, could you send us a link to a recent build showing the issues you are still having. Please send it to support [at] travis-ci [dot] com.

Thanks!

**Comment by guoyingtao** [comment]
@cotsog Just sent a email to support email. Thanks for your help!

**Comment by guoyingtao** [comment]
@cotsog With the help of support from travis-ci.dom, I finally made it out. Thanks again.

The reason is travis-ci default "script" command for iOS projects is currently outdated i.e. it's using `xctool` and this utility has deprecated building iOS projects since Xcode 8.
So I changed to xcodebuild, it worked finally.
- set -o pipefail && xcodebuild -project YouProject -scheme YourScheme -destination 'platform=iOS Simulator,name=iPhone 6 Plus,OS=10.1' build test | xcpretty

**Comment by vtkumaresan** [comment]
what is the conclusion for this issue, i'm getting same issue, please suggest a solution,
Fix WWDRCA Certificate
Unable to delete certificate matching "0950B6CD3D2F37EA246A1AAA20DFAADBD6FE1F75"security: AppleWWDRCA.cer: already in /Library/Keychains/System.keychain
